wsf_scraping/pipelines.py

```diff
@@ -25,63 +26,58 @@ def __init__(self):
 
         self.logger = logging.getLogger(__name__)
         self.logger.setLevel(logging.INFO)
+        self.database = SQLite3Connector()
 
         spider_loader = spiderloader.SpiderLoader.from_settings(self.settings)
         spiders = spider_loader.list()
 
         for spider_name in spiders:
             os.makedirs('./results/pdfs/%s' % spider_name, exist_ok=True)
 
-    def process_item(self, item, spider):
-        """Process items sent by the spider."""
-
+    def check_keywords(self, item, spider_name):
         keep_pdf = self.settings['KEEP_PDF']
         download_only = self.settings['DOWNLOAD_ONLY']
         feed = self.settings['FEED_CONFIG']
-        pdf_path = ''.join(['./results/pdfs/', spider.name, '/'])
+        pdf_path = ''.join(['./results/pdfs/', spider_name, '/'])
 
         # Convert PDF content to text format
-        f = open('/tmp/' + item['pdf'], 'rb')
+        with open('/tmp/' + item['pdf'], 'rb') as f:
+            if download_only:
+                os.rename(
+                    ''.join(['/tmp/', item['pdf']]),
+                    ''.join([pdf_path, item['pdf']])
+                )
+                return item
 
-        if download_only:
-            os.rename(
-                ''.join(['/tmp/', item['pdf']]),
-                ''.join([pdf_path, item['pdf']])
-            )
-            return item
+            self.logger.info('Processing: %s (%s)' % (
```

Comment:
minor: don't interpolate strings with the logging module, just pass them as extra args to the logging call, i.e.

```python
self.logger.info('Processing: %s (%s)', item['pdf'], self.settings['PARSING_METHOD'])
```

If the log isn't sent anywhere, you save the cost of interpolating the string.